### DISCUSSION TITLE
[Bug report] [Confirmed] can't use apostrophe (single quote) in tag titles

### DISCUSSION THREAD

**Discussion by hadsh** [GetPublii/Publii#1861] [Bug Report]
### Operating system

win 11

### Publii version

0.46.2 (build 16968)

### Issue type

Application

### Bug description

I can't use apostrophe ' (also call single quote) in a tag title. I suppose **it's not escape** ?  I didn't re-test it, I just avoid the single quote, but it's not the best solution :) Thx for all the work on publii !

### Steps to reproduce

just try create a tag titled :"  No Man's Sky " for example :) tag a few posts, and come back in the app, to the tags page. you may see that now you have 2 tags with the same name

### Relevant log output
edit to add the screenshot : 

![tags-wrong](https://github.com/user-attachments/assets/edad5e1a-c3ca-4ef2-9012-6371af1df731)

**Comment by michaelfluegge** [comment]
I tested your tag under macOS Sonoma, and it worked as expected. Still one tag "No Man's Sky", only, after restarting the app.

**Reply by hadsh** [reply]
I edit, and update the report. Yes, Publii misbehave with tags including a single quote

**Comment by hadsh** [comment]
it's Back ! Unfortunately, the bug is still very real !
 
It happens after multiple upload, and changes into my tags.

as you see in the screenshot, there is 2 "Tutorials No Man's Sky" since I add the single quote, and publii misbehave not knowing which is which; the number of posts link to each tags is different, so it's really a **small but bad bug**

 
![tags-wrong](https://github.com/user-attachments/assets/c553cd99-8001-4697-bfa3-14518d87391c)

**Comment by dziudek** [comment]
@hadsh - I have really tried but I am unable to reproduce this issue :( I have tried many ways - create tag and then assign it to posts. Create tags during post creation. Create one tag with `'` and another with `\`` - in all cases all is working fine.

Could you provide step by step flow which leads to this issue? I need an instruction which I am able to reproduce to diagnose and fix the reason of this issue. I believe that it can happen, but currently I do not know why and how :/

**Comment by hadsh** [comment]
Hello ! Sorry for the late reply. My theory is the characters aren't properly escape ?

When I use ' straight simple quote ' (the programmer's quote, Unicode+0027) or  "  straight double quote " (programmer's double quote, U+0022), directly from keyboard, these are the same we used in code, the tags became unstable

I start using the typographic ’closing single quote’ (U+2019) or « guillemet »  (U+00AB/BB) which are, in fact, better :) and it works fine

nevertheless, I simply use my programmer's quote or double quote in the tag name (without \escaping them) tag a few posts, and when I came back to the tags page, I got a dupe tag (without the same number of posts attributed, so it's bad). I had to change the original tag's name and delete the duplicate, re-tag everything :) 

I can't really explain better, it's just normal use, no fancy tricks. On the screenshot above, I got the same problem for each of the tags with a programmer's quote or programmer's double quote

I hope it help :) 

and Thx for Publii !

**Reply by hadsh** [reply]
could be the same here :

https://github.com/GetPublii/Publii/issues/1742
